catalog/catalog-client/src/main/java/com/cgdpd/library/catalog/client/rest/LibraryCatalogHttpClient.java

```diff
@@ -27,4 +30,19 @@ public Mono<DetailedBookDto> getDetailedBookDto(Isbn13 isbn13) {
               .bodyToMono(DetailedBookDto.class)
               .onErrorMap(onErrorMap());
     }
+
+    @Override
+    public Flux<DetailedBookDto> searchBooks(PaginationCriteria paginationCriteria,
+                                             SearchBookCriteria searchBookCriteria) {
```

Comment:
This is not being used why?

Update both Criteria records to have a method

`public MultiValueMap<String, String> toQueryParams()` create interface with that and use it to create the uri